File: digitalmaktabspa/src/models/School.ts
import { AddAddressDto } from "../dtos/AddAddressDto";
import { AddPhoneNumberDto } from "../dtos/AddPhoneNumberDto";
import { SchoolType } from "./SchoolType";
import { UserRole } from "./UserRole";

export interface School {
  userRole: UserRole;
  schoolName: string;
  address: AddAddressDto;
  phoneNumber: AddPhoneNumberDto;
  email: string;
  password: string;
  confirmPassword?: string;
  logo?: File | null;
  code: string;
  schoolType: SchoolType;
  schoolTypeValue: string;
}

File: digitalmaktabapi/Models/School.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace digitalmaktabapi.Models
{
    [Table("School")]
    public class School : Base
    {
        public required string SchoolName { get; set; }
        public required Address Address { get; set; }
        public required PhoneNumber PhoneNumber { get; set; }
        public required string Email { get; set; }
        public required byte[] PasswordHash { get; set; }
        public required byte[] PasswordSalt { get; set; }
        public string? Logo { get; set; }
        public required int Code { get; set; }
        public required SchoolType SchoolType { get; set; }
        public required UserRole UserRole { get; set; }
        public ICollection<Student> Students { get; set; } = [];
        public ICollection<Branch> Branches { get; set; } = [];
    }
}